fontforge/splinefit.c

```diff
@@ -731,248 +731,315 @@ return( SplineMake3(from,to));
 	from->nextcp = from->me; to->prevcp = to->me;
 return( SplineMake3(from,to));
     }
+    /* This is the generic case, where a generic part is approximated by a cubic */
+    /* bezier spline. */
+	if ( is_accurate ) { /* More accurate but slower function by Linus Romer */
```

Comment:
I think I *basically* understand this, and would characterize it roughly as "brute force based on limits already calculated for the other algorithm".

One thing I don't understand is that I would expect the "fine-tuning" run to search in a narrower range of measures than the "narrowing" run. However, I'm having a little trouble understanding the motivation for what seems to amount to `ftlen*fromunit.x*((31*fromhandle)/30.0)/60.0` in run 0. Is the idea more or less that this is close enough to `ftlen*fromunit.x*fromhandle/60` that it's not worth two separate statements or is there something more subtle going on?

Reply:
Well, "brute force based on limits already calculated for the other algorithm" is fine for me.

I do not understand how you got `ftlen*fromunit.x*((31*fromhandle)/30.0)/60.0`. If `run==0` the part should read as `ftlen*fromunit.x*( fromhandle )/60.0`.

While I see that this part is not well readable, it is much more compact than a more readable version. I will try to explain it for fromhandle:
run 0:
fromhandle runs from 1 to 60, the handle is divided in parts from 1/60,2/60,...,60/0. The best is chosen, say 45/60.
run 1:
fromhandle runs from -30 to 30, the handle is divided in parts from 44/60, (45-29/30)/60, ..., 45/60, ... , (45+29/30)/60, 46/60

But I agree, that fromhandle in run 1 should only take values from -29 to 29.

Reply:
I suspect I was just confused in that I was getting my ternary operator precedence wrong.

There's a school that advocates only adding parentheses when they're needed and a different school according to which it's good to add redundant parentheses for cases less frequently encountered. I suppose you're using spacing to echo the precedence but the FontForge source base does unusual things with spacing so that may do much in context.